### PR TITLE
Jaccard recommendation

### LINKED ISSUE #46: Improve recommendation
- Enhance recommendations using Jaccard Distance
- Add test case
- Recommend books from multiple genre

### PR DESCRIPTION
**Pull request by phucla0122**
Improve the recommendation feature with Jaccard distance algorithm
close #46

### CODE REVIEW

File: src/test/java/com/groupseven/sysc4806project/HttpTest.java [hunk 1]
```diff
@@ -200,6 +200,7 @@ public void recommendation() throws Exception {
         when(userService.createUser(name, false)).thenReturn(1);
         this.mockMvc.perform(get("/user/recommendation").cookie(new Cookie("userId", userId.toString()))
                         .param("username", name))
-                .andExpect(status().isFound());
+                .andExpect(status().isFound())
+                .andExpect(content().string(containsString("")));
```
Comment (nicolastuttle) [inline src/test/java/com/groupseven/sysc4806project/HttpTest.java]: Checking for empty string is equivalent to not checking at all. I think this should be reworked to make sure a specific book appears

Comment (phucla0122) [reply src/test/java/com/groupseven/sysc4806project/HttpTest.java]: I'm having some trouble with that, do you have any suggestions regarding this?

Comment (nicolastuttle) [reply src/test/java/com/groupseven/sysc4806project/HttpTest.java]: If you limit the recommendation page to let's say 5 books, you could create 6 books, manually compute the Jaccard distance for each, and ensure that the 5 closest show up in recommendations. Not sure if you can check the order

Comment (phucla0122) [reply src/test/java/com/groupseven/sysc4806project/HttpTest.java]: I see, I'll try to do it then

**Review by nicolastuttle — APPROVED** [review]

**Review by robellgabriel — APPROVED** [review]

**Pull request merged**